libcloudforensics/providers/azure/internal/account.py

```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Represents an Azure account."""
+
+from time import sleep
+from typing import Optional, Dict
+
+# pylint: disable=import-error
+from azure.mgmt.compute import ComputeManagementClient
+from azure.mgmt.compute.v2016_04_30_preview.models import DiskCreateOption
+from msrestazure.azure_exceptions import CloudError
+# pylint: enable=import-error
+
+from libcloudforensics.providers.azure.internal import compute, common
+
+
+class AZAccount:
+  """Class that represents an Azure Account.
+
+  Attributes:
+    subscription_id (str): The Azure subscription ID to use.
+    credentials (ServicePrincipalCredentials): An Azure credentials object.
+    compute_client (ComputeManagementClient): An Azure compute client object.
+  """
+
+  def __init__(self, subscription_id: str) -> None:
+    """Initialize the AZAccount class.
+
+    Args:
+      subscription_id (str): The Azure subscription ID to use.
+    """
+    self.subscription_id = subscription_id
+    self.credentials = common.GetCredentials()
+    self.compute_client = ComputeManagementClient(
+        self.credentials, self.subscription_id)
+
+  def ListInstances(
+      self,
+      resource_group_name: Optional[str] = None) -> Dict[str, compute.AZVirtualMachine]:  # pylint: disable=line-too-long
+    """List instances in an Azure subscription / resource group.
+
+    Args:
+      resource_group_name (str): Optional. The resource group name to list
+          instances from. If none specified, then all instances in the AZ
```

Comment:
It's best to use the full name of the service when referred to in docstrings.

```suggestion
          instances from. If none specified, then all instances in the Azure
```